msteams-platform/m365-apps/actions-in-m365.md

```diff
@@ -0,0 +1,213 @@
+
+Update the handler to receive the Action information through the context object to create a seamless user experience for performing users specific tasks using the [Teams JavaScript library (TeamsJS)](/javascript/api/@microsoft/teams-js).
+
+When a user selects Add option from the app's context menu, a page or dialog opens with the help of the `openDialog` property in the app manifest. Your app can access contextual information about the invoked Action from the `actionInfo` property of the context object `app.getContext()`.
+
+```javascript
+const context = await app.getContext();
+const itemId = context.actionInfo && context.actionInfo.actionObjects[0].itemId;
+this.setState({
+    itemId: itemId
+});
+```
+
+**Parameters**
+
+| &nbsp; | Name | Description |
+| --- | --- | --- |
+| &nbsp; | `actionId` | Maps to the action id supplied inside the manifest. |
+| &nbsp; | `actionObjects` | Array of corresponding action objects. |
+
+* [`actionInfo`](/javascript/api/@microsoft/teams-js/actioninfo): The TeamsJS helps to enable your app to determine when a user opens a page or dialog from an Action, and the content that initiated the Action.
+
+    ```javascript
+    app.getContext().then((context: app.Context) => {
+        const actionInfo = context.actionInfo;
+        if (actionInfo && actionInfo.actionId == 'myActionId1') {
+            // Handle specific action
+        }
+    });
+    ```
+
+* `itemId`: The app receives the id as the content and uses it to query the Microsoft graph.
+
+#### Access content through Graph API
+
+After obtaining the `itemId` of the triggering content, you can use the Graph API to read or modify the content, facilitating task completion for your users.
+
+```javascript
+async readActionItem() {
+    try {
+        return await this.graphClient.api(`/users/${this.objectId}/drive/items/${this.itemId}`).get();
+    } catch (error) {
+        console.log("readActionItem", error);
+    }
+}
+```
+
+## Build and run your app
+
+After you update the app manifest, you can build and run your app in Teams Toolkit.
+
+To build and run your app locally:
+
+1. Open your app in Teams Toolkit.
+
+1. From the left pane, select **Debug in the Microsoft 365 app (Edge) without backend** and enter **F5**.
+
+   :::image type="content" source="images/actions-debug.png" alt-text="The screenshot shows actions in debug.":::
+
+1. Select **Yes** if the following dialog appears:
+
+   :::image type="content" source="~/assets/images/teams-toolkit-v2/first-tab/hw-warning.png" alt-text="Screenshot shows the microsoft warning.":::
+
+A browser window opens with Microsoft 365 home page and your app is available under **Apps**.
+
+You can now preview your Actions in Microsoft 365 app, right-click a file that is supported by your Actions. Actions appear in the context menu, for example **Add todo task**.
+
+:::image type="content" source="images/actions-context-menu.png" alt-text="The screenshot shows the actions in context menu.":::
+
```

Comment:
We're missing a section here on how to distribute your app across your tenant via Microsoft Admin Center (in private preview doc)